test/unit-test/FreeRTOS_ND/FreeRTOS_ND_utest.c

```diff
@@ -0,0 +1,2024 @@
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * http://aws.amazon.com/freertos
+ * http://www.FreeRTOS.org
+ */
+
+
+/* Include Unity header */
+#include "unity.h"
+
+/* Include standard libraries */
+#include <stdlib.h>
+#include <string.h>
+#include <stdint.h>
+#include "FreeRTOS.h"
+
+#include "mock_task.h"
+#include "mock_list.h"
+
+/* This must come after list.h is included (in this case, indirectly
+ * by mock_list.h). */
+#include "mock_queue.h"
+#include "mock_event_groups.h"
+
+#include "mock_FreeRTOS_ARP.h"
+#include "mock_FreeRTOS_IP.h"
+#include "mock_FreeRTOS_IPv6.h"
+#include "mock_FreeRTOS_IP_Private.h"
+#include "mock_FreeRTOS_IP_Timers.h"
+#include "mock_FreeRTOS_IP_Utils.h"
+#include "mock_FreeRTOS_IPv6_Utils.h"
+#include "mock_FreeRTOS_Routing.h"
+#include "mock_NetworkBufferManagement.h"
+
+#include "catch_assert.h"
+#include "FreeRTOS_ND_stubs.c"
+#include "FreeRTOS_ND.h"
+
+/** @brief The ND cache. */
+extern NDCacheRow_t xNDCache[ ipconfigND_CACHE_ENTRIES ];
+
+/* Setting IPv6 address as "fe80::7009" */
+static const IPv6_Address_t xDefaultIPAddress =
+{
+    0xfe, 0x80, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x70, 0x09
+};
+
+/* IPv6 multi-cast address is ff02::. */
+static const IPv6_Address_t xMultiCastIPAddress =
+{
+    0xff, 0x02, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00
+};
+
+/* Setting eIPv6_SiteLocal IPv6 address as "feC0::7009" */
+static const IPv6_Address_t xSiteLocalIPAddress =
+{
+    0xfe, 0xC0, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x70, 0x09
+};
+
+/* Setting IPv6 Gateway address as "fe80::1" */
+static const IPv6_Address_t xGatewayIPAddress =
+{
+    0xfe, 0x80, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x01
+};
+
+/* IPv6 default MAC address. */
+static const MACAddress_t xDefaultMACAddress = { 0x22, 0x22, 0x22, 0x22, 0x22, 0x22 };
+
+#define xHeaderSize                                   ( ipSIZE_OF_ETH_HEADER + ipSIZE_OF_IPv6_HEADER + sizeof( ICMPHeader_IPv6_t ) )
+
+/** @brief When ucAge becomes 3 or less, it is time for a new
+ * neighbour solicitation.
+ */
+#define ndMAX_CACHE_AGE_BEFORE_NEW_ND_SOLICITATION    ( 3U )
+
+
+static BaseType_t NetworkInterfaceOutputFunction_Stub_Called = 0;
```

Comment:
This should be Type INT, since BaseType_t generally associated with pdTRUE/pdFALSE ?